manager/base.py

```diff
@@ -665,9 +667,30 @@ def parse_tags(tags_list):
     return tags
 
 
+def inventory_groups_condition(group_ids) -> str:
```

Comment:
This whole function can be reimplemented using the `Expression` and `fn` classes from peewee.
For example:
`InventoryHosts.select().where(Expression(InventoryHosts.groups, "@>", fn.ANY(SQL("'{}'::jsonb"))))`

This way, the SQL injection is not possible, thus, we do not need the regex checking.